karton/yaramatcher/yaramatcher.py

```diff
@@ -97,17 +97,15 @@ def args_parser(cls):
         return parser
 
     @classmethod
-    def main(cls):
-        parser = cls.args_parser()
-        args = parser.parse_args()
+    def config_from_args(cls, config: Config, args: argparse.Namespace) -> None:
+        super().config_from_args(config, args)
+        config.load_from_dict({"yaramatcher": {"rules": args.rules}})
 
-        config = Config(args.config_file)
-        service = YaraMatcher(config=config, yara_rule_dir=args.rules)
-        service.loop()
-
-    def __init__(self, yara_rule_dir: Optional[str] = None, *args, **kwargs) -> None:
+    def __init__(self, *args, **kwargs) -> None:
         super().__init__(*args, **kwargs)
-        self.yara_handler = YaraHandler(path=yara_rule_dir or "rules")
+        self.yara_handler = YaraHandler(
+            path=self.config.get("yaramatcher", "rules", "rules")
```

Comment:
```suggestion
            path=self.config.get("yaramatcher", "rules", fallback="rules")
```

Small fix to be clearer (the repetition of "rules" might confuse some people)